### DISCUSSION TITLE
issue: where are the messages for basic validation

### DISCUSSION THREAD

**Discussion by MrOxMasTer** [react-hook-form/react-hook-form#11730] [Q&A]
### Version Number

7.51.2

### Codesandbox/Expo snack

https://codesandbox.io/p/devbox/heuristic-cache-dh5gnp?file=%2Fapp%2Fpage.tsx%3A6%2C42&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clucgl21z00073o6hfqkuidqy%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clucgl21z00023o6h6nrsz9tl%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clucgl21z00043o6h3f2ulzhw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clucgl21z00063o6h8zkvvuwv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B63.4562103318858%252C36.5437896681142%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clucgl21z00023o6h6nrsz9tl%2522%253A%257B%2522id%2522%253A%2522clucgl21z00023o6h6nrsz9tl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucgmm9u00br3o6hrb14bfy1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fapp%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A6%252C%2522startColumn%2522%253A42%252C%2522endLineNumber%2522%253A6%252C%2522endColumn%2522%253A42%257D%255D%257D%255D%252C%2522activeTabId%2522%253A%2522clucgmm9u00br3o6hrb14bfy1%2522%257D%252C%2522clucgl21z00063o6h8zkvvuwv%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucgl21z00053o6hto0xe1xi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clucgl21z00063o6h8zkvvuwv%2522%252C%2522activeTabId%2522%253A%2522clucgl21z00053o6hto0xe1xi%2522%257D%252C%2522clucgl21z00043o6h3f2ulzhw%2522%253A%257B%2522id%2522%253A%2522clucgl21z00043o6h3f2ulzhw%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucgl21z00033o6h07f1zsrp%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clucgl21z00033o6h07f1zsrp%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to reproduce

1. Click on any of the form fields
2. To see that the inscription is missing

### Expected behaviour

At least some kind of inscription appeared when required, if you don't set your own

### What browsers are you seeing the problem on?

Chrome

There was such a problem and I couldn't figure out what it was. 

I use the "input" forms in my own component and use the "??" operator for the label. The problem is, I had to output a lot to the console to figure out where the error lies, because the inscription just disappears. 

It turns out that the errors object from form State for some reason assigns an empty string to required when an error occurs, from which the operator "??" thinks that this is normal and outputs nothing to me.

Why is there no base label when required? Well, or either do undefined so that such situations do not occur. Because an empty line makes no sense, and it adds problems, as it is now. An unknown behavior appears

I know you can add your validation both with zod and in required in the video object in the message field, but it seems as if it delivers undefined behaviour

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sujinleeme** [comment]
I can't access the codesandbox(devbox) because of an infinite loop.

https://github.com/react-hook-form/react-hook-form/assets/6011401/acb73d53-4cfb-42af-b862-5f7b3955ad96

Could you check the Field component?
